### PR TITLE
backend tests: GET management/v2/devices

### PR DESCRIPTION
**Pull request by mchalski**
https://tracker.mender.io/browse/MEN-2529

note it's on a new `device_filtering` branch, mirroring inventory's `device_filtering` - where the GET is actually implemented.

a lot of this was hunting time-sensitive bugs around conductor (see comments in code).
also - some optimization, I started with data setup per test func, which caused massive overhead (couple mins total). had to be reworked to saner per-class setup.

tested manually with https://github.com/mendersoftware/inventory/pull/137 and https://github.com/mendersoftware/mender-api-gateway-docker/pull/104

### CODE REVIEW

File: backend-tests/tests/test_inventory.py [hunk 1]
```diff
@@ -307,6 +336,476 @@ def test_fail_no_attr_value(self, user):
...
+                                        inventory_v1.URL_DEVICE_ATTRIBUTES,
                                         payload)
             assert r.status_code == 400
+
+def make_devs_get_v2(utoken, devauthd, invd, invi, tenant_token='', tenant_id=''):
+    """ GET v2 - specific data fixture 
+        Create 20 accepted devices, with inventory.
+        All submit:
+        - inventory via PATCH v1/devices/inventory/devices.
+        - id data via PATCH v1/internal/inventory/devices.
```
Comment (kjaskiewiczz) [inline backend-tests/tests/test_inventory.py]: v2

File: backend-tests/tests/test_inventory.py [hunk 2]
```diff
@@ -14,31 +14,48 @@
...
 
 from api.client import ApiClient
-from common import mongo, clean_mongo
+from common import mongo, clean_mongo, clean_mongo_cls
 from infra.cli import CliUseradm, CliDeviceauth, CliTenantadm
 import api.deviceauth as deviceauth_v1
 import api.useradm as useradm
-import api.inventory as inventory
+import api.inventory as inventory_v1
+import api.inventory_v2 as inventory
```
Comment (kjaskiewiczz) [inline backend-tests/tests/test_inventory.py]: I can't see inventory_v2.py in api directory.

Comment (mchalski) [reply backend-tests/tests/test_inventory.py]: fixed

File: backend-tests/tests/test_inventory.py (hunk 1 above)
Comment (mchalski) [reply backend-tests/tests/test_inventory.py]: fixed

**Pull request merged**